rclcpp/include/rclcpp/client.hpp

```diff
@@ -694,10 +753,28 @@ class Client : public ClientBase
     CallbackWithRequestTypeValueVariant>;
 
   int64_t
-  async_send_request_impl(const Request & request, CallbackInfoVariant value)
+  async_send_request_impl(SharedRequest request, CallbackInfoVariant value)
   {
+    if (use_intra_process_) {
+      auto ipm = weak_ipm_.lock();
+      if (!ipm) {
+        throw std::runtime_error(
+                "intra process send called after destruction of intra process manager");
+      }
+      bool intra_process_server_available = ipm->service_is_available(intra_process_client_id_);
+
+      if (intra_process_server_available) {
```

Comment:
why if the server is not available we fall back to DDS comm?

Reply:
Because the server might be available in another process, or created with IPC OFF.

Reply:
Good point.
We should add a comment about it.

Reply:
Done in 3e4d9c65